### ISSUE TITLE
CUDD crash on Win64 with JDK9

### ISSUE THREAD

**Issue by kleinj** [prismmodelchecker/prism#13]
As reported in #12, PRISM GUI does not startup properly on Windows with a JDK9 early release snapshot. The 32bit version of PRISM works fine, but the 64bit version crashes quite early during CUDD initialisation in dd.dll (both xprism and standard prism). JDK8 works fine.

Some initial debugging indicates that this might be related to pointer arithmetic that CUDD uses to store complement edge bits inside `DdNode` pointers. If that is the only problem, it should be easy to fix.

**Comment by kleinj** [comment]
It looks indeed to be a problem with pointer arithmetic in CUDD. Windows 64bit uses the  [IL32P64](https://en.wikipedia.org/wiki/64-bit_computing#64-bit_data_models) data model, where `int` and `long` are 32 bit but pointers are 64 bit. In contrast, Linux/OS X uses I32LP64, i.e., `int` at 32 bit and `long` and pointers at 64 bit.

The CUDD version currently in PRISM (2.5.0) assumes that it can convert from pointers to `long` and back (e.g. for storing a single bit in the least significant bit of a `DdNode` pointer to indicate a complemented edge). On Windows 64 this leads to clearing the upper 32 bits of the pointer value. If the upper 32 bit of the pointer are non-zero, the pointer is garbled, most likely points to some invalid memory region and results in a hard crash when trying to read/write to that location.

Curiously, when using JDK8 and the Win64 binary of PRISM, the test suite runs without a crash. This indicates that the pointers allocated by CUDD have a zero upper half there, at least for the tested executions / models / properties. With JDK9, this behaviour seems to have changed and we tend to get pointers with non-zero upper half immediately on CUDD startup, which will crash PRISM.

It looks like the current CUDD release [3.0.0](http://vlsi.colorado.edu/~fabio/) has fixed the pointer arithmetic by switching to `uintptr_t` as the integer data type for the bit fiddling. So, once we have imported this version, the crashes with JDK9 should hopefully be fixed as well.

There was one location in the PRISM native code parts where mingw compilation for Win64 warned about invalid pointer conversions, which has been fixed in prismmodelchecker/prism-svn@559c970ef796b4d686b2179507ab0b42987f62d6.
